File: practice/algorithms/search/ice_cream_parlor/IceCreamParlor.java
import java.util.*;

public class IceCreamParlor {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int tests = Integer.parseInt(stdin.nextLine());
        for(int i = 0; i < tests; i++) {
            int amount = Integer.parseInt(stdin.nextLine());
            int length = Integer.parseInt(stdin.nextLine());
            int[] costs = new int[length];
            String[] strings = stdin.nextLine().split(" ");
            for(int j = 0; j < length; j++) {
                costs[j] = Integer.parseInt(strings[j]);
            }
            for(int j = 0; j < length - 1; j++) {
                for(int k = j + 1; k < length; k++) {
                    if(costs[j] + costs[k] == amount) {
                        System.out.println((j + 1) + " " + (k + 1));
                    }
                }
            }
        }
        stdin.close();
    }
}

File: practice/algorithms/search/ice_cream_parlor/IceCreamParlor.scala
import scala.collection.mutable.ListBuffer
import scala.io.Source

object IceCreamParlor extends App {
  val console = Source.stdin.bufferedReader()
  val tests = console.readLine().toInt
  (0 until tests).foreach(_ => {
    val amount = console.readLine().toInt
    val lenght = console.readLine().toInt
    val costs = console.readLine().split(" ").map(_.toInt)
    println(indexes(costs, amount).mkString(" "))
  })

  def indexes(costs: Array[Int], amount: Int): List[Int] = {
    val result: ListBuffer[Int] = ListBuffer.empty
    (0 until costs.length - 1).foreach(i => {
      (i + 1 until costs.length).foreach(j => {
        if(costs(i) + costs(j) == amount) {
          result += i + 1
          result += j + 1
        }
      })
    })
    result.toList
  }
}